guides/runtime-parser.md

```diff
@@ -0,0 +1,323 @@
+
+```html
+<!DOCTYPE html>
+<html lang="en"><head>
+  <meta charset="utf-8">
+    <style>
+      /* App styles */
+    </style>
+    <style>
+      /* Material toolbar styles */
+    </style>
+    <style>
+      /* Material circle styles */
+    </style>
+  </head>
+<body>
+  <hello-mobile-app _nghost-gcc-1="">
+    <md-toolbar _ngcontent-gcc-1=""><div class="md-toolbar-layout"> <md-toolbar-row>
+      <div _ngcontent-gcc-1="" class="icon ng"></div>
+      Hello Mobile
+     </md-toolbar-row>  </div></md-toolbar>
+    <!--template bindings={}--><!--shellRender(<md-spinner _ngcontent-gcc-1="" mode="indeterminate" role="progressbar" _nghost-gcc-3=""> <svg _ngcontent-gcc-3="" preserveAspectRatio="xMidYMid meet" viewBox="0 0 100 100"> <circle _ngcontent-lqi-3="" cx="50px" cy="50px" r="40px" style="stroke-dashoffset:251.3274;"></circle> </svg> </md-spinner>)-->
+    <!--template bindings={}--><h1 _ngcontent-gcc-1="" shellNoRender="">App is Fully Rendered</h1>
+  </hello-mobile-app>
+
+  <script src="/node_modules/core-js/client/shim.min.js?1468224624471"></script>
+  <script src="/node_modules/systemjs/dist/system.src.js?1468224624493"></script>
+  <script src="./system-config.js"></script>
+
+  <script src="/node_modules/zone.js/dist/zone.js?1468224624495"></script>
+  <script src="/node_modules/rxjs/bundles/Rx.js?1468224624496"></script>
+
+  <script>
+  // Application initialization
+  System.import('app/main')
+    .catch(function (e) {
+      console.error(e,
+        'Report this error at https://github.com/mgechev/angular2-seed/issues');
+    });
+  </script>
+</body>
+</html>
+```
+
+Notice the following:
+
+- Inside the `hello-mobile-app` element the App Shell of the application is wrapped inside `<!--shellRender(...)-->`.
+- The elements which are not supposed to be part of the Application Shell are annotated with the `shellNoRender` attribute (note that the attributes are case sensitive).
+
+Once we invoke the `parseDoc` method with argument a response which as body has the template above, the following actions will be performed:
+
+- All the referenced within the template images that match any of the file extensions defined in `INLINE_IMAGES` will be inlined as base64 strings.
+- All the elements annotated with `shellNoRender` attribute will be stripped.
+- The content of all `<!--shellRender(...)-->` comments will be used as part of the Application Shell.
+
+## Step 3 - Handling the Fetch Event
+
+As final step, lets see how our Service Worker is going to handle the `fetch` event:
+
+```html
+self.addEventListener('fetch', function (event: any) {
+  event.respondWith(
+    ngShellParser.match(event.request)
+      .then((response: any) => {
+        if (response) return response;
+        return context.caches.match(event.request)
+          .then((response: any) => {
+            if (response) {
+              return response;
+            }
```

Comment:
Nitpick: This is formatted differently than above (`if (response) return response;`).